sklearn/utils/estimator_checks.py

```diff
@@ -1721,3 +1722,25 @@ def check_classifiers_regression_target(name, Estimator):
     e = Estimator()
     msg = 'Unknown label type: '
     assert_raises_regex(ValueError, msg, e.fit, X, y)
+
+
+@ignore_warnings(category=DeprecationWarning)
+def check_decision_proba_consistency(name, Estimator):
+    # Check whether an estimator having both decision_function and
+    # predict_proba methods has outputs with perfect rank correlation.
+
+    rnd = np.random.RandomState(0)
+    X_train = (2*rnd.uniform(size=(10, 4))).astype(int)
+    y = X_train[:, 0]
```

Comment:
I don't think we want this so strongly correlated with the input. A decent classifier will ignore the other feature and so the decision function will be very peaked. For this test to be meaningful we want the decision function to be highly variable, and hence I think `y` should be as uncorrelated with `X` as possible.